File: tools/api_handshake.js
const https = require('https');

const testBarcode = '7891000053508'; // Coca-cola
const url = `https://world.openfoodfacts.org/api/v0/product/${testBarcode}.json`;

console.log(`[*] Testando conexão com Open Food Facts: ${testBarcode}...`);

https.get(url, (res) => {
    let data = '';
    res.on('data', (chunk) => { data += chunk; });
    res.on('end', () => {
        try {
            const json = JSON.parse(data);
            if (json.status === 1) {
                console.log(`[SUCCESS] Produto encontrado: ${json.product.product_name} (${json.product.brands})`);
                process.exit(0);
            } else {
                console.log('[INFO] API respondeu, mas produto não encontrado. Conectividade OK.');
                process.exit(0);
            }
        } catch (e) {
            console.error(`[ERROR] Falha ao parsear JSON: ${e.message}`);
            process.exit(1);
        }
    });
}).on('error', (err) => {
    console.error(`[FATAL] Erro de conexão: ${err.message}`);
    process.exit(1);
});

File: tools/api_handshake.py
import requests
import sys

def test_openfoodfacts(barcode):
    print(f"[*] Testando código de barras: {barcode}")
    url = f"https://world.openfoodfacts.org/api/v0/product/{barcode}.json"
    try:
        response = requests.get(url, timeout=10)
        if response.status_code == 200:
            data = response.json()
            if data.get('status') == 1:
                product = data.get('product', {})
                name = product.get('product_name', 'N/A')
                brand = product.get('brands', 'N/A')
                print(f"[SUCCESS] Produto encontrado: {name} ({brand})")
                return True
            else:
                print("[INFO] Produto não encontrado na base OFF.")
                return True # API respondeu, teste passou
        else:
            print(f"[ERROR] API retornou status {response.status_code}")
            return False
    except Exception as e:
        print(f"[FATAL] Erro na conexão: {e}")
        return False

if __name__ == "__main__":
    # Exemplo: Coca-cola (7891000053508)
    test_barcode = "7891000053508"
    if test_openfoodfacts(test_barcode):
        sys.exit(0)
    else:
        sys.exit(1)
